src/library_wasmfs_fetch.js

```diff
@@ -73,17 +109,39 @@ addToLibrary({
       // read/getSize fetch the data, then forward to the parent class.
       read: async (file, buffer, length, offset) => {
         try {
-          await getFile(file);
+          await getFileRange(file, offset || 0, length);
         } catch (response) {
           return response.status === 404 ? -{{{ cDefs.ENOENT }}} : -{{{ cDefs.EBADF }}};
         }
-        return jsFileOps.read(file, buffer, length, offset);
+        var fileInfo = wasmFS$JSMemoryRanges[file];
+        var fileData = fileInfo.chunks;
+        var chunkSize = fileInfo.chunkSize;
+        var firstChunk = (offset / chunkSize) | 0;
+        var lastChunk = ((offset+length) / chunkSize) | 0;
+        if(offset + length > lastChunk * chunkSize) {
+          lastChunk += 1;
+        }
+        var readLength = 0;
+        for (var i = firstChunk; i < lastChunk; i++) {
+          var chunk = fileData[i];
+          var start = Math.max(i*chunkSize, offset);
+          if(!chunk) {
+            throw [fileData.length, firstChunk, lastChunk, i];
+          }
+          var chunkStart = i*chunkSize;
+          var end = Math.min(chunkStart+chunkSize, offset+length);
+          HEAPU8.set(chunk.subarray(start-chunkStart, end-chunkStart), buffer+(start-offset));
+          readLength = end - offset;
+        }
+        return readLength;
       },
       getSize: async (file) => {
         try {
-          await getFile(file);
-        } catch (response) {}
-        return jsFileOps.getSize(file);
+          await getFileRange(file, 0, 0);
+        } catch (response) {
```

Comment:
Should this be called `error` or something other then response?  Do we really want to just ignore this error and return 0?   (I guess that is simply the existing behaviour).

Reply:
I called it failedResponse, I don't love throwing things that aren't errors but I didn't want to make too many changes.